.github/resource/js/login.js

```diff
@@ -0,0 +1,65 @@
+const form = document.querySelector('#login_form');
+
+// 이메일 input
+const email = document.querySelector('#email');
+const error = document.querySelector('.error_text');
+const error_pwd = document.querySelector('.password_text');
+const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+// 비밀번호 input
+const password = document.querySelector('#password');
+
+// 링크
+const hostname = '/folder';
+
+// 비밀번호 아이콘
+const pwd_icon = document.querySelector('.pwd_icon')
+
+email.addEventListener('focusout', function(e) {
+    if(email.value == '') {
+        error.style.display = 'block';
+        email.classList.add('error');
+        error.innerText = "이메일을 입력해 주세요";
+    } else if(!emailRegex.test(email.value)) {
+        error.style.display = 'block';
+        email.classList.add('error');
+        error.innerText = "올바른 이메일 주소가 아닙니다.";
+    } else if(emailRegex.test(email.value) && email.value !== '') {
+        error.style.display = 'none';
+        email.classList.remove('error');
+    }
```

Comment:
에러를 표출하는 showError, 제거하는 hideError 함수를 분리해 재활용하면 좋을 것 같습니다.
그리고 에러 관리는 객체로 하면 관리하기 더 편하겠네요!

예를 들면,
```js
const ERROR_MESSAGE = {
  email_empty: '이메일을 입력해 주세요.',
  email_invalid: '올바른 이메일 주소가 아닙니다.',
  pwd_empty: ...,
  ...
}
```
이런 식으로 관리하고
```js
const showError = (targetEl, errorEl, errorType) => {
  errorEl.style.display = 'block';
  targetEl.classList.add('error');
  errorEl.innerText = ERROR_MESSAGE[errorType];
}
```
대강.. 이런 식으로 만든 다음
```suggestion
    if(email.value == '') {
        showError(email, error, ERROR_MESSAGE['email_empty']
    } else if(!emailRegex.test(email.value)) {
        ...
    } else if(emailRegex.test(email.value) && email.value !== '') {
        ...
    }
```
요런 방식으로 사용해볼 수 있을 것 같네요.
물론 간략하게 나눈 코드긴 하지만, 데이터 관리하기도 용이하고 에러 출력 형식을 한꺼번에 쉽게 바꿀 수 있을 것 같습니다.

Reply:
그리고 `==` 대신 엄격 비교 연산자 `===`를 활용해서 값과 타입까지 체크하시는 게 좋습니다.
단순 동등 연산자만을 사용하면 타입을 변환해 비교하기 때문에, 예상치 못한 오류를 만날 수 있습니다.
대신 타입까지 체크하는 `엄격 비교 연산자`를 사용해 주세요!